### PR TITLE
Hot fix for the race condition problem

### PR DESCRIPTION
**Pull request by rayshih**
Detail: https://github.com/auth0/react-native-auth0/issues/83

### CODE REVIEW

**Comment by cocojoe** [comment]
Thanks for the PR, will take a look.

**Comment by cocojoe** [comment]
@lbalmaceda can you take a look at this as well please.

File: android/src/main/java/com/auth0/react/A0Auth0Module.java [hunk 1]
```diff
@@ -125,13 +127,19 @@ String generateCodeChallenge(@NonNull String codeVerifier) {
...
+                Callback cb = A0Auth0Module.this.callback;
+                if (cb != null) {
+                    final WritableMap error = Arguments.createMap();
+                    error.putString("error", "a0.session.user_cancelled");
+                    error.putString("error_description", "User cancelled the Auth");
+                    cb.invoke(error);
+                    A0Auth0Module.this.callback = null;
+                }
+            }
+        }, 100);
```
Comment (cocojoe) [inline android/src/main/java/com/auth0/react/A0Auth0Module.java]: Why this value?

Comment (rayshih) [reply android/src/main/java/com/auth0/react/A0Auth0Module.java]: just a random guess

**Comment by Blackening999** [comment]
@cocojoe @rayshih 
how long will it take to fix this PR?
and what exact issue is?

This item spoiling the experience for my users :(

I'd love to help if I know exactly what's the issue

**Comment by cocojoe** [comment]
@Blackening999 this is being discussed, thanks for your patience.

File: android/src/main/java/com/auth0/react/A0Auth0Module.java (hunk 1 above)
Comment (hzalaz) [reply android/src/main/java/com/auth0/react/A0Auth0Module.java]: Use a constant for this one please

Comment (hzalaz) [inline android/src/main/java/com/auth0/react/A0Auth0Module.java]: Use a constant please

Comment (rayshih) [reply android/src/main/java/com/auth0/react/A0Auth0Module.java]: Do you mean like this?

```java
public static int DELAY = 100;
```

Comment (hzalaz) [reply android/src/main/java/com/auth0/react/A0Auth0Module.java]: sort of, add a `private static final int CANCEL_EVENT_DELAY` and we are good to merge

**Comment by cocojoe** [comment]
Applied change in #100

**Pull request closed without merging**